homeassistant/components/cmus/media_player.py

```diff
@@ -86,11 +86,38 @@ def __init__(self, server, password, port, name):
             self.cmus = remote.PyCmus()
             auto_name = "cmus-local"
         self._name = name or auto_name
+        self._server = server
+        self._password = password
+        self._port = port
         self.status = {}
 
+    def reconnect(self):
```

Comment:
This reconnect method invokes `remote.PyCmus()` within a try block, allowing it to fail. `__init__` executes the same code without a try block. Probably they should use the same code so the integration doesn't fail to start if the first connection attempt fails, and for less code duplication.

Reply:
Good call - just tidied this up and it feels a bit tighter now.

Reply:
I'd move this method to a new class that when instantiated and connected has the cmus attribute. Then pass an instance of this new class to the entity when creating it.